### PR TITLE
[1880] auction, draft and foreign investors

### PR DESCRIPTION
**Pull request by bentziaxl**
Added auction round for privates
Added foreign investors
Added stubs around Beijing
added foreign investors draft

### CODE REVIEW

File: lib/engine/game/g_1880/game.rb [hunk 1]
```diff
@@ -167,6 +167,42 @@ class Game < Game::Base
...
+
+        def next_round!
+          @round =
+            case @round
+            when Round::Stock
+              @operating_rounds = @phase.operating_rounds
+              reorder_players
+              new_operating_round
+            when Round::Operating
+              if @round.round_num < @operating_rounds
```
Comment (crericha) [inline lib/engine/game/g_1880/game.rb:188]: Just a placeholder for now?

File: lib/engine/game/g_1880/step/selection_auction.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../../step/passable_auction'
+
+module Engine
+  module Game
+    module G1880
+      module Step
+        class SelectionAuction < Engine::Step::Base
```
Comment (crericha) [inline lib/engine/game/g_1880/step/selection_auction.rb:10]: I believe selection auction already exists

File: lib/engine/game/g_1880/step/simple_draft.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1880
+      module Step
+        class SimpleDraft < Engine::Step::Base
```
Comment (crericha) [inline lib/engine/game/g_1880/step/simple_draft.rb:9]: Drafts already exist too (1846, 18CZ, etc.)

File: lib/engine/game/g_1880/step/selection_auction.rb (hunk 2 above)
Comment (crericha) [reply lib/engine/game/g_1880/step/selection_auction.rb:10]: I looked at the code and I see that so far selection auction and draft only exist within games. Is this selection auction or the simple draft quite different from the others? If not, please add to `lib/engine/steps` and subclass for any minor differences. If they are quite different, then carry on.

File: lib/engine/game/g_1880/game.rb (hunk 1 above)
Comment (bentziaxl) [reply lib/engine/game/g_1880/game.rb:188]: Yeah, i took the standard next_round! code and only modified to make the auction and draft rounds work.
this code will change a lot to make it work with 1880 round order/SR rules

File: lib/engine/game/g_1880/step/selection_auction.rb (hunk 2 above)
Comment (bentziaxl) [reply lib/engine/game/g_1880/step/selection_auction.rb:10]: they are pretty similar. Will subclass

**Pull request merged**